File: JobApplicationTracker.Api/DTOs/JAObjects/JobApplications/JobApplicationMinimalDto.cs
using JobApplicationTracker.DTOs.JAEvent;
using JobApplicationTracker.Enums;

namespace JobApplicationTracker.DTOs;

public class JobApplicationMinimalDto
{
    public Guid JAId { get; set; }
    public string Company { get; set; }
    public string Position { get; set; }
    public JAStatusType JAStatus { get; set; }
    public JAEventType? EventType { get; set; }
    public DateTime? EventDate  { get; set; }
    public bool? IsWholeDay { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: JobApplicationTracker.Frontend/types/JAObjects/JobApplications/JobApplicationMinimalDto.ts
import type { JAStatusType } from '../../Enums/JAStatusType';
import type { JAEventType } from '../../Enums/JAEventType';

export interface JobApplicationMinimalDto {
  jaId: string;
  company: string;
  position: string;
  jaStatus: JAStatusType;
  eventType: JAEventType | null;
  eventDate: string | null;
  isWholeDay: boolean | null;
  updatedAt: string;
}